pontoon/base/views.py

```diff
@@ -456,6 +457,82 @@ def get_team_comments(request):
     return JsonResponse(payload, safe=False)
 
 
+def _send_add_comment_notifications(user, comment, entity, locale, translation):
+    # On translation comment, notify:
+    #   - authors of other translation comments in the thread
+    #   - translation author
+    #   - translation reviewers
+    if translation:
+        recipients = list(translation.comments.values_list("author__pk", flat=True))
+
+        recipients.append(translation.user.pk)
+
+        if translation.approved_user:
+            recipients.append(translation.approved_user.pk)
+        if translation.unapproved_user:
+            recipients.append(translation.unapproved_user.pk)
+        if translation.rejected_user:
+            recipients.append(translation.rejected_user.pk)
+        if translation.unrejected_user:
+            recipients.append(translation.unrejected_user.pk)
+
+    # On team comment, notify:
+    #   - project-locale translators or locale translators
+    #   - locale managers
+    #   - authors of other team comments in the thread
+    #   - authors of translation comments
+    #   - translation authors
+    #   - translation reviewers
+    else:
+        recipients = []
+        project_locale = ProjectLocale.objects.get(
+            project=entity.resource.project, locale=locale,
+        )
+        translations = Translation.objects.filter(entity=entity, locale=locale)
+
+        # Only notify translators of the project if defined
+        translators = project_locale.translators_group.user_set.values_list(
+            "pk", flat=True
+        )
+        if len(translators) == 0:
+            translators = locale.translators_group.user_set.values_list("pk", flat=True)
+
+        recipients.append(translators)
+        recipients.append(locale.managers_group.user_set.values_list("pk", flat=True))
+
+        recipients.append(
+            Comment.objects.filter(entity=entity, locale=locale).values_list(
+                "author__pk", flat=True
+            )
+        )
+
+        recipients.append(
+            Comment.objects.filter(translation__in=translations).values_list(
+                "author__pk", flat=True
+            )
+        )
+
+        recipients.append(translations.values_list("user__pk", flat=True))
+        recipients.append(translations.values_list("approved_user__pk", flat=True))
+        recipients.append(translations.values_list("unapproved_user__pk", flat=True))
+        recipients.append(translations.values_list("rejected_user__pk", flat=True))
+        recipients.append(translations.values_list("unrejected_user__pk", flat=True))
+
+        recipients = [item for sublist in recipients for item in sublist]
+
+    for recipient in (
+        User.objects.filter(pk__in=recipients).exclude(pk=user.pk).distinct()
```

Comment:
If you make `recipients` a `set`, you can remove the call to `distinct` here. It might also be worth removing the current user's pk from the `recipients` set directly, instead of filtering them out in SQL.